gcp/gcp.go

```diff
@@ -0,0 +1,124 @@
+package gcp
+
+import (
+	"container/list"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+	google "golang.org/x/oauth2/google"
+	compute "google.golang.org/api/compute/v1"
+	"net/http"
+	"time"
+)
+
+type GcpContext struct {
+	Client  *http.Client
+	Service *compute.Service
+	Project string
+	Regions []*compute.Region
+}
+
+func DefaultContext() (*GcpContext, error) {
+	creds, err := google.FindDefaultCredentials(oauth2.NoContext, compute.ComputeScope)
+	if err != nil {
+		return nil, err
+	}
+	client := oauth2.NewClient(context.Background(), creds.TokenSource)
+
+	service, err := compute.New(client)
+	if err != nil {
+		return nil, err
+	}
+
+	regions, err := service.Regions.List(creds.ProjectID).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	context := &GcpContext{
+		Client:  client,
+		Service: service,
+		Project: creds.ProjectID,
```

Comment:
I wonder if it's possible to specify the project id and nuke multiple projects instead of fetching the default one?